c/src/reduce.cu

```diff
@@ -0,0 +1,866 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cub/detail/choose_offset.cuh>
+#include <cub/grid/grid_even_share.cuh>
+#include <cub/util_device.cuh>
+
+#include <cuda/std/cstdint>
+#include <cuda/std/functional>
+
+#include <format>
+#include <iostream>
+#include <memory>
+
+#include <cccl/reduce.h>
+#include <nvJitLink.h>
+#include <nvrtc.h>
+
+void check(nvrtcResult result)
```

Comment:
Nitpick: `check` is a rather nondescript word for a function that potentially throws an exception?

I would rather have something like `assert` or similar